From d7204ee7bb542a0ffc776bc88e3c7c447fbb4b0b Mon Sep 17 00:00:00 2001
From: Sudarshan N Ramachandra <sudarshan.n.ramachandra@intel.com>
Date: Tue, 17 Feb 2015 17:20:57 +0530
Subject: [PATCH 130/441] pci-ish: Register suspend/resume callbacks for ISH

When platform enters S3 (mwait S3), bus mastering
of ISH will be disabled. In this state if ISH
firmware tries to do a DMA, it will hang.

In order to avoid bus mastering disable/enable
done by PCI core during suspend/resume, Need to
register dummy suspend/resume handlers in ISH
driver

Change-Id: I926cfd3c0360a802160d7d559d81ffcf5fd0f993
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-14867
Signed-off-by: Sudarshan N Ramachandra <sudarshan.n.ramachandra@intel.com>
Reviewed-on: https://android.intel.com:443/333201
(cherry picked from commit b270eff14d5250cc0f7d28fefcf4f6651df1eb42)
---
 drivers/misc/heci/pci-ish.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

--- a/drivers/misc/heci/pci-ish.c
+++ b/drivers/misc/heci/pci-ish.c
@@ -691,7 +691,23 @@ static void ish_remove(struct pci_dev *p
 	pci_disable_device(pdev);
 }
 
-#define HECI_ISH_PM_OPS	NULL
+#define HECI_ISH_PM_OPS	(&intel_ish_pm)
+
+
+static int intel_ish_suspend(struct device *dev)
+{
+	return 0;
+}
+
+static int intel_ish_resume(struct device *dev)
+{
+	return 0;
+}
+
+static const struct dev_pm_ops intel_ish_pm = {
+	.suspend = intel_ish_suspend,
+	.resume = intel_ish_resume,
+};
 
 /*
  *  PCI driver structure
